tests/dialects/test_dialect.py

```diff
@@ -1399,7 +1399,8 @@ def test_cross_join(self):
             write={
                 "drill": "SELECT a, b FROM x CROSS JOIN UNNEST(y, z) AS t(a, b)",
                 "presto": "SELECT a, b FROM x CROSS JOIN UNNEST(y, z) AS t(a, b)",
-                "spark": "SELECT a, b FROM x LATERAL VIEW EXPLODE(y) t AS a LATERAL VIEW EXPLODE(z) t AS b",
+                "spark": "SELECT a, b FROM x LATERAL VIEW INLINE(ARRAYS_ZIP(y, z)) t AS a, b",
+                "hive": "SELECT a, b FROM x LATERAL VIEW EXPLODE(y) t AS a LATERAL VIEW EXPLODE(z) t AS b",
```

Comment:
It looks like the transpilation for unnest with >1 expressions was wrong all along? cc @tobymao

```
trino> WITH x AS (SELECT ARRAY[1,2,3] AS y, ARRAY[4,5,6] AS z) SELECT a, b FROM x CROSS JOIN UNNEST(y, z) AS t (a, b);
 a | b 
---+---
 1 | 4 
 2 | 5 
 3 | 6 
(3 rows)
```

Current SQLGlot main produces:

```python
(.venv) ➜  sqlglot git:(main) python
>>> import sqlglot
>>> sqlglot.transpile("WITH x AS (SELECT ARRAY[1,2,3] AS y, ARRAY[4,5,6] AS z) SELECT a, b FROM x CROSS JOIN UNNEST(y, z) AS t (a, b)", "trino", "spark")
['WITH x AS (SELECT ARRAY(1, 2, 3) AS y, ARRAY(4, 5, 6) AS z) SELECT a, b FROM x LATERAL VIEW EXPLODE(y) t AS a LATERAL VIEW EXPLODE(z) t AS b']
```

And Spark yields different results:

```
spark-sql (default)> WITH x AS (SELECT ARRAY(1, 2, 3) AS y, ARRAY(4, 5, 6) AS z) SELECT a, b FROM x LATERAL VIEW EXPLODE(y) t AS a LATERAL VIEW EXPLODE(z) t AS b;
a	b
1	4
1	5
1	6
2	4
2	5
2	6
3	4
3	5
3	6
Time taken: 4.504 seconds, Fetched 9 row(s)
spark-sql (default)> WITH x AS (SELECT ARRAY(1, 2, 3) AS y, ARRAY(4, 5, 6) AS z) SELECT a, b FROM x LATERAL VIEW INLINE(ARRAYS_ZIP(y, z)) t AS a, b;
a	b
1	4
2	5
3	6
Time taken: 0.633 seconds, Fetched 3 row(s)
```

----

The hive case is still incorrect right? Having two `LATERAL VIEW EXPLODE` clauses will result in a cross join instead of an array zip & join. We don't have to fix it now if it's not trivial, but I suggest we get rid of this (misleading) test, and then do `self.unsupported` for Hive if there are multiple expressions to avoid data corruption issues.

Reply:
Yes, as you can see. I delete this line of test case